client/src/app/components/stardust/history/transactionHistoryUtils.ts

```diff
@@ -0,0 +1,98 @@
+import { INodeInfoBaseToken, OutputResponse } from "@iota/sdk-wasm/web";
+import moment from "moment/moment";
+
+import { ITransactionHistoryItem } from "~models/api/stardust/ITransactionHistoryResponse";
+import { IOutputDetailsMap } from "~helpers/hooks/useAddressHistory";
+import { TransactionsHelper } from "~helpers/stardust/transactionsHelper";
+import { CHRYSALIS_MAINNET } from "~models/config/networkType";
+import { DateHelper } from "~helpers/dateHelper";
+import { formatAmount } from "~helpers/stardust/valueFormatHelper";
+
+export const groupOutputsByTransactionId = (historyView: ITransactionHistoryItem[], outputDetailsMap: IOutputDetailsMap) => {
+    const byTransactionId = new Map<string, (OutputResponse & ITransactionHistoryItem)[]>();
+    historyView.forEach((historyItem) => {
+        const outputDetails = outputDetailsMap[historyItem.outputId];
+        if (!outputDetails) {
+            return;
+        }
+        const transactionId = historyItem.isSpent ?
+            outputDetails.metadata.transactionIdSpent :
+            outputDetails.metadata.transactionId;
+
+        if (!transactionId) {
+            return;
+        }
+
+        if (!byTransactionId.has(transactionId)) {
+            byTransactionId.set(transactionId, []);
+        }
+
+        const transaction = byTransactionId.get(transactionId);
+        transaction?.push({...historyItem, ...outputDetails});
+    });
+    return byTransactionId;
+}
+
+export const calculateBalanceChange = (outputs: (OutputResponse & ITransactionHistoryItem)[]) => {
+    return outputs.reduce((acc, output) => {
+        if (output.isSpent) {
+            return acc - Number(output.output.amount);
+        }
+        return acc + Number(output.output.amount);
+    }, 0);
+};
+
+export interface ITransactionHistoryRecord {
+    isGenesisByDate: boolean;
+    isTransactionFromStardustGenesis: boolean;
+    isSpent: boolean;
+    transactionLink: string;
+    transactionId: string;
+    timestamp: number;
+    dateFormatted: string;
+    balanceChange: number;
+    balanceChangeFormatted: string;
+    outputs: (OutputResponse & ITransactionHistoryItem)[];
+}
+
+export const getTransactionHistoryRecords = (
+    transactionIdToOutputs: Map<string, (OutputResponse & ITransactionHistoryItem)[]>,
+    network: string,
+    tokenInfo: INodeInfoBaseToken,
+    isFormattedAmounts: boolean
+) => {
```

Comment:
Let's add explicit return type here, and move this and the interface `ITransactionHistoryRecord` at the top of the file.